src/main/java/org/broadinstitute/hellbender/tools/exome/allelefraction/AlleleFractionModeller.java

```diff
@@ -0,0 +1,363 @@
+import org.broadinstitute.hellbender.tools.exome.SegmentedModel;
+import org.broadinstitute.hellbender.utils.mcmc.AdaptiveMetropolisSampler;
+import org.broadinstitute.hellbender.utils.mcmc.GibbsSampler;
+import org.broadinstitute.hellbender.utils.mcmc.ParameterizedModel;
+import org.broadinstitute.hellbender.utils.mcmc.Sampler;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static java.lang.Math.log;
+import static java.lang.Math.sqrt;
+import static org.broadinstitute.hellbender.utils.MathUtils.logFactorial;
+
+/**
+ * Given a {@link org.broadinstitute.hellbender.tools.exome.SegmentedModel} and counts of alt and ref reads over a list of het sites,
+ * infers the minor allele fraction of each segment.  For example, a segment
+ * with (alt,ref) counts (10,90), (11,93), (88,12), (90,10) probably has a minor allele fraction
+ * somewhere around 0.1.  The model takes into account allelic bias due to mapping etc. by learning
+ * a global gamma distribution on allelic bias ratios.
+ *
+ * For a given het locus, we define the bias ratio to be the expected ratio of
+ * mapped ref reads to mapped alt reads given equal amounts of DNA (that is, given
+ * a germline het).  The model learns a common gamma distribution:
+ *      bias ratio ~ Gamma(mu*beta, beta)
+ * where mu is the global mean of bias ratios (probably slightly larger than 1.00), and
+ * beta is the rate parameter such that variance = mu/beta.
+ *
+ * Each segment has a minor allele fraction f, and for each het within the locus
+ * the number of alt reads is drawn from a binomial distribution with total count
+ * n = #alt reads + #ref reads and alt probability f/(f + (1-f)*bias ratio) if the
+ * locus is alt minor and (1-f)/(1-f + f*bias ratio) if the locus is ref minor.
+ *
+ * Conceptually, the model contains latent variable corresponding to the bias ratio
+ * and an indicator for alt minor/ref minor at each het locus.  However, we integrate them
+ * out and the MCMC model below only contains the minor allele fractions and
+ * the three hyperparameters of the model: the two parameters of the gamma distribution
+ * along with the global outlier probability.
+ *
+ * See docs/AllelicCapSeg/ACS-methods.pdf for a thorough description of the model.
+ *
+ * @author David Benjamin &lt;davidben@broadinstitute.org&gt;
+ */
+public final class AlleleFractionModeller {
+    private final ParameterizedModel<AlleleFractionState, AlleleFractionData> model;
+    private final List<Double> meanBiasSamples = new ArrayList<>();
+    private final List<Double> biasVarianceSamples = new ArrayList<>();
+    private final List<Double> outlierProbabilitySamples = new ArrayList<>();
+    private final List<AlleleFractionState.MinorFractions> minorFractionsSamples = new ArrayList<>();
+    private final int numSegments;
+
+    // INNER SAMPLER CLASSES -------------------------------------------------------------------------------------------
```

Comment:
rather than make these classes, i think it saves quite a bit of code to just make them static Sampler instances.  for example:

    @VisibleForTesting
    protected static final Sampler<Double, AlleleFractionState, AlleleFractionData> meanBiasSampler = (rng, state, dataCollection) -> {
        final AdaptiveMetropolisSampler sampler = new AdaptiveMetropolisSampler(state.meanBias(), INITIAL_MEAN_BIAS_STEP_SIZE, 0, Double.POSITIVE_INFINITY);
        return sampler.sample(x -> logLikelihood(state.shallowCopyWithProposedMeanBias(x), dataCollection));
    };

where `INITIAL_MEAN_BIAS_STEP_SIZE` can be extracted as a private static constant.  then in the unit test, you can simply use

`final Sampler<Double, AlleleFractionState, AlleleFractionData> meanBiasSampler = AlleleFractionModeller.meanBiasSampler;`

this obviates the need for constructors, etc.